binnacle/lib/providers/navigation_provider.dart

```diff
@@ -1,65 +1,284 @@
+  PolarPlot _plot;
+
+  double _currentIdealHeading;
+  BehaviorSubject<IdealHeadingModel> idealHeading;
+
+  int _currentCheckPoint;
+
+  NavigationEvent _currentEvent;
+  BehaviorSubject<NavigationEvent> eventBus;
   ReplaySubject<PositionModel> positionHistory;
 
-  List<LatLng> course;
+  
+  double _closeEnough = (1 / metersPerDegree) * defaultCloseEnough; // 25 meters
+  double get closeEnough => _closeEnough;
+
+  
+  double _maxOffset = (1 / metersPerDegree) * defaultMaxOffset; // 25 meters
+  double get maxOffset => _maxOffset;
 
-  NavigationProvider({@required position, @required wind}) {
+  NavigationProvider(
+      {@required Stream<PositionModel> position,
+      @required BehaviorSubject<WindModel> wind,
+      PolarPlot plot}) {
     _position = position;
     _wind = wind;
-
-    eventBus.add(NavigationEvent(eventType: NavigationEventType.init));
+    print("Printing plot!");
+    print(plot);
+    _plot = plot;
+    eventBus = BehaviorSubject<NavigationEvent>(sync: true);
+    idealHeading = BehaviorSubject<IdealHeadingModel>();
   }
 
-  start(LatLng start, LatLng end) {
-    /// NOTE: naive to assume long = x, lat = y, but it's close enough
-    /// A better implementation would utilize the Haversine formula
-    /// Which IMO should be done in the algorithm
+  /// Used to initialize new course
+  /// Args:
+  ///   (LatLng) start: Start of the course, usually the current location
+  ///   (LatLng) end: Where the user wants to go.
+  /// Returns:
+  ///   Nothing, async
+  Future start(LatLng start, LatLng end) async {
+    if (_plot == null) {
+      // Set up polar plot
+      print("Plot was null!");
+      await _initPolarPlot();
+    }
+    _updateEventBus(NavigationEventType.calculatingRoute);
     _start = Vector2(start.longitude, start.latitude);
     _end = Vector2(end.longitude, end.latitude);
+    print("About to get wind!");
+    // Gets most recent wind entry
+    _windModel = await _wind.firstWhere((w) {
+      return (w != null);
+    });
+    print("Got wind!");
+    _initRoute(_plot, _start, _end, _windModel);
+    print("About to get position history");
+    /// Record the position history now that we have a course
+    positionHistory = new ReplaySubject<PositionModel>();
 
-    eventBus
-        .add(NavigationEvent(eventType: NavigationEventType.calculatingRoute));
+    _currentCheckPoint = 0;
+    print("About to start listener");
+    _positionSub = await _position.listen(_navigate);
+  }
 
-    _route = new RouteModel(start: _start, end: _end, wind_radians: 0);
-    // new RouteModel(start: _start, end: _end, wind_radians: currentWind.deg);
+  /// Sets close enough with the meter as units. Will be converted to lat/long degrees
+  /// Args:
+  ///   (double) meters : Close enough for finish or tacks in meters
+  /// Returns:
+  ///   void
+  void setCloseEnough(double meters) {
+    if (meters < 0) {
+      // Don't accept negative numbers
```

Comment:
This should throw an exception if it's expecting negative numbers to be passed in. That or we check against the absolute value.

Reply:
Added a throw FormatException, meant to do this eventually, but forgot